src/tree.ts

```diff
@@ -22,7 +22,7 @@ export class Tree {
    * @param {boolean} [isBranch] - An option that makes a branch from created tree. Branch can have children.
    */
   public constructor(node: TreeModel, parent: Tree = null, isBranch: boolean = false) {
-    this.buildTreeFromModel(node, parent, isBranch);
+    this.buildTreeFromModel(node, parent, isBranch || Boolean(node.children));
```

Comment:
In this case
```
node.children = "hello"
Boolean(node.children)
```

will also return `true`. Maybe it should be more restrictive - `Array.isArray(node.children)`?